src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1697,6 +1714,32 @@ private bool IsMiniShell()
             return false;
         }
 
+        /// <summary>
+        /// Returns true if native command is powershell or pwsh and the experimental feature is enabled.
+        /// </summary>
+        /// <returns></returns>
+        private bool CheckNativeAppIsNotPowerShell()
+        {
+            if (!ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeScriptBlockArgumentFeatureName))
+            {
+                return false;
+            }
+
+            // We have to check if the command is pwsh or powershell.
+            // This will still miss the case where the command is a hard-link to pwsh.
+            PSObject possibleLink = new PSObject(new FileInfo(Command.CommandInfo.Source));
+            string resolvedPath = Microsoft.PowerShell.Commands.InternalSymbolicLinkLinkCodeMethods.ResolvedTarget(possibleLink);
```

Comment:
```suggestion
            string resolvedPath =  File.ResolveLinkTarget(Command.CommandInfo.Source, returnFinalTarget: true)?.FullName;

```

Reply:
@iSazonov your suggestion seems to only work if the file is actually a symbolic link.

Reply:
Ah, my bad.